automata/fa/dfa.py

```diff
@@ -266,14 +266,13 @@ def _validate_transitions(self, start_state: DFAStateT, paths: DFAPathT) -> None
         self._validate_transition_invalid_symbols(start_state, paths)
         self._validate_transition_end_states(start_state, paths)
 
-    def validate(self):
+    def validate(self) -> None:
```

Comment:
@eliotwrobson Do we need to explicitly declare the return type for the `validate` methods on the subclasses? I would think the return type would be inherited from the superclass (`Automaton`, in this case).

Reply:
I don't think this is technically necessary, but I think we should strive to have annotations on as many functions as possible (not sure if there's precedent for that anywhere but typecheckers include flags that disallow untyped functions).